### PR TITLE
Fixed failing monitoring installation on OCP4

### PR DESCRIPTION
**Pull request by robshelly**
### Type of change

- Bugfix

### Description

Changed systemtests to use existing monitoring stack on OpenShift4

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by robshelly** [comment]
@enmasse-ci run tests ocp4 testcase=MonitoringTest

**Comment by robshelly** [comment]
@enmasse-ci run tests crc testcase=MonitoringTest

**Comment by robshelly** [comment]
> @robshelly not sure I understand these changes, why you disabled tests on ocp3? And also why only on CRC ? :)

I've fixed this so the tests are not disabled on OCP3.

Also, it's not configured to only use CRC, it's just checking if it's running on CRC. The test fails when there is already some monitoring infra installed on the cluster (the installation of the Application Monitoring Operator fails) so this PR is to fix that. The fix is the following:

* OCP < 4: No existing monitoring on cluster so install Application Monitoring Operator
* CRC: Prometheus operator is installed but in an 'unmanaged' state and therefore scaled to zero. The easiest workaround for this is to uninstall the existing Prometheus and install the Application Monitoring Operator
* OCP4: Just use the existing monitoring stack

**Comment by kornys** [comment]
@enmasse-ci run tests profile=acceptance testcase=MonitoringTest test_only crc

**Comment by lulf** [comment]
@robshelly Probably outside the scope of this PR, but going forward should we start relying on the already installed prometheus operator on OCP4 and remove our example-dependency on the application-monitoring-operator?

File: systemtests/src/test/java/io/enmasse/systemtest/isolated/monitoring/MonitoringTest.java [hunk 1]
```diff
@@ -47,14 +48,26 @@
...
+
+        }
+
+        if (!Kubernetes.isOpenShiftCompatible(OCP4) || Kubernetes.isCRC()) {
+            try { //TODO remove it after upgrade to surefire plugin 3.0.0-M5
+                EnmasseOperatorManager.getInstance().installMonitoringOperator();
+                Endpoint prometheusEndpoint = Kubernetes.getInstance().getExternalEndpoint("prometheus-route", environment.getMonitoringNamespace());
+                monitoring = new MonitoringClient(prometheusEndpoint);
+                monitoring.waitUntilPrometheusReady();
+                kubernetes.createNamespace(testNamespace);
```
Comment (kornys) [inline systemtests/src/test/java/io/enmasse/systemtest/isolated/monitoring/MonitoringTest.java:67]: This line must run in every case, so move it to the beginning of this beforeall method.

File: systemtests/src/test/java/io/enmasse/systemtest/isolated/monitoring/MonitoringTest.java [hunk 2]
```diff
@@ -69,7 +82,9 @@ void catchBeforeAllException() throws Exception {
     void uninstallMonitoring() throws Exception {
         beforeAllException = null; //TODO remove it after upgrade to surefire plugin 3.0.0-M5
         EnmasseOperatorManager.getInstance().removeIoT();
-        EnmasseOperatorManager.getInstance().deleteMonitoringOperator();
+        if (Kubernetes.isCRC()) {
```
Comment (kornys) [inline systemtests/src/test/java/io/enmasse/systemtest/isolated/monitoring/MonitoringTest.java:85]: I think we should also call uninstall on ocp3.11?

**Comment by kornys** [comment]
@robshelly I have just tried to run this branch on ocp4 and it does not work. If I install enmasse and open prometheus UI which is by default on ocp4 I all enmasse queries returns no data. And also test uses route from prometheus deployed by enmasse. So if we want to use prometheus from ocp4 we have to add switch into code which change route for MonitoringClient etc...

**Comment by robshelly** [comment]
@enmasse-ci run tests profile=acceptance testcase=MonitoringTest test_only crc

**Comment by robshelly** [comment]
@enmasse-ci run tests profile=acceptance testcase=MonitoringTest test_only crc

**Comment by robshelly** [comment]
@enmasse-ci run tests profile=acceptance testcase=MonitoringTest test_only crc

File: systemtests/src/test/java/io/enmasse/systemtest/isolated/monitoring/MonitoringTest.java [hunk 3]
```diff
@@ -150,22 +161,29 @@ void testAddressQueries() throws Exception {
                 .build();
 
         resourcesManager.setAddresses(false, topic, queue);
+        String infraUuid = addressSpace.getAnnotation(AnnotationKeys.INFRA_UUID);
+        System.out.println("Infra UUID::" + infraUuid);
```
Comment (kornys) [inline systemtests/src/test/java/io/enmasse/systemtest/isolated/monitoring/MonitoringTest.java:165]: please use LOGGER instead of System.out.println

**Comment by kornys** [comment]
@robshelly looks good thank you please fix my comment and I will merge it

File: systemtests/src/test/java/io/enmasse/systemtest/isolated/monitoring/MonitoringTest.java (hunk 3 above)
Comment (robshelly) [reply systemtests/src/test/java/io/enmasse/systemtest/isolated/monitoring/MonitoringTest.java]: :+1: Fixed

**Pull request merged**